src/app/pages/courses-page/courses-page.component.spec.ts

```diff
@@ -1,25 +1,82 @@
+import { Component, Input } from '@angular/core';
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 
+import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
+
+import { Course } from '../../interfaces/course.interface';
+import { CoursesService } from '../../services/courses.service';
+import { MockCoursesService } from '../../services/courses.service.mock';
 import { CoursesPageComponent } from './courses-page.component';
 
+@Component({ selector: 'app-search-input' })
+class StubSearchInputComponent { }
+
+@Component({ selector: 'app-course-item' })
+class StubCourseItemComponent {
+  @Input() course: Course;
+}
+
+@Component({ selector: 'app-load-more-panel' })
+class StubLoadMorePanelComponent { }
+
 describe('CoursesPageComponent', () => {
   let component: CoursesPageComponent;
   let fixture: ComponentFixture<CoursesPageComponent>;
+  let coursesService: CoursesService;
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      declarations: [ CoursesPageComponent ],
+      imports: [ FontAwesomeModule ],
+      declarations: [
+        CoursesPageComponent,
+        StubSearchInputComponent,
+        StubCourseItemComponent,
+        StubLoadMorePanelComponent,
+      ],
+      providers: [ { provide: CoursesService, useValue: MockCoursesService } ],
```

Comment:
👍

Reply:
you can make it an exported provider:
`export const mockCoursesServiceProvider = { provide: CoursesService useClass:  MockCoursesService };`

Reply:
Good point, thanks